process/smartContract/processorV2/processV2.go

```diff
@@ -0,0 +1,2645 @@
+	if err != nil {
+		log.Trace("Processing error", "error", err.Error())
+		return vmcommon.ExecutionFailed, sc.ProcessIfError(acntSnd, txHash, tx, err.Error(), []byte(vmOutput.ReturnMessage), snapshot, vmInput.GasLocked)
+	}
+
+	acntSnd, err = sc.reloadLocalAccount(acntSnd) // nolint
+	if err != nil {
+		log.Debug("reloadLocalAccount error", "error", err.Error())
+		return 0, err
+	}
+	finalResults, logsFromSCRs := sc.cleanInformativeOnlySCRs(results)
+
+	vmOutput.Logs = append(vmOutput.Logs, logsFromSCRs...)
+	err = sc.updateDeveloperRewards(tx, vmOutput, 0)
+	if err != nil {
+		log.Debug("updateDeveloperRewardsProxy", "error", err.Error())
+		return 0, err
+	}
+
+	err = sc.scrForwarder.AddIntermediateTransactions(finalResults)
+	if err != nil {
+		log.Debug("AddIntermediate Transaction error", "error", err.Error())
+		return 0, err
+	}
+
+	totalConsumedFee, totalDevRwd := sc.computeTotalConsumedFeeAndDevRwd(tx, vmOutput, 0)
+	sc.txFeeHandler.ProcessTransactionFee(totalConsumedFee, totalDevRwd, txHash)
+	sc.printScDeployed(vmOutput, tx)
+	sc.gasHandler.SetGasRefunded(vmOutput.GasRemaining, txHash)
+
+	sc.vmOutputCacher.Put(txHash, vmOutput, 0)
+
+	ignorableError := sc.txLogsProcessor.SaveLog(txHash, tx, vmOutput.Logs)
+	if ignorableError != nil {
+		log.Debug("scProcessor.DeploySmartContract() txLogsProcessor.SaveLog()", "error", ignorableError.Error())
+	}
+
+	return 0, nil
+}
+
+func (sc *scProcessor) printScDeployed(vmOutput *vmcommon.VMOutput, tx data.TransactionHandler) {
+	scGenerated := make([]string, 0, len(vmOutput.OutputAccounts))
+	for _, account := range vmOutput.OutputAccounts {
+		if account == nil {
+			continue
+		}
+
+		addr := account.Address
+		if !core.IsSmartContractAddress(addr) {
+			continue
+		}
+
+		scGenerated = append(scGenerated, sc.pubkeyConv.Encode(addr))
+	}
+
+	log.Debug("SmartContract deployed",
+		"owner", sc.pubkeyConv.Encode(tx.GetSndAddr()),
+		"SC address(es)", strings.Join(scGenerated, ", "))
+}
+
+// taking money from sender, as VM might not have access to him because of state sharding
+func (sc *scProcessor) processSCPayment(tx data.TransactionHandler, acntSnd state.UserAccountHandler) error {
+	if check.IfNil(acntSnd) {
+		// transaction was already processed at sender shard
+		return nil
+	}
+
+	acntSnd.IncreaseNonce(1)
+	err := sc.economicsFee.CheckValidityTxValues(tx)
+	if err != nil {
+		return err
+	}
+
+	cost := sc.economicsFee.ComputeTxFee(tx)
+	cost = cost.Add(cost, tx.GetValue())
+
+	if cost.Cmp(big.NewInt(0)) == 0 {
+		return nil
+	}
+
+	err = acntSnd.SubFromBalance(cost)
+	if err != nil {
+		return err
+	}
+
+	return nil
+}
+
+func (sc *scProcessor) processVMOutput(
+	vmInput *vmcommon.VMInput,
+	vmOutput *vmcommon.VMOutput,
+	txHash []byte,
+	tx data.TransactionHandler,
+	acntSnd state.UserAccountHandler,
+	snapshot int,
+) ([]data.TransactionHandler, error) {
+
+	createdAsyncCallback, scrTxs, err := sc.processSCOutputAccounts(vmInput, vmOutput, tx, txHash)
```

Comment:
on v1 we did not call this. Why calling it here?

Reply:
This was called in the same `processVMOutput()` function in V1 (line 1848)